src/main/resources/avro/bdg.avdl

```diff
@@ -87,6 +87,47 @@ record RecordGroupMetadata {
   union { null, int } predictedMedianInsertSize = null;
   union { null, string } platform = null;
   union { null, string } platformUnit = null;
+
+  /**
+   The IDs of the processing steps that have been applied to this record group/
+   */
+  array<string> processingStepIds = [];
```

Comment:
Since this `*..*` association is between `RecordGroupMetadata`, `ProcessingStep` (, and later `Sample`), all of which will not be large in number, why not have a direct reference instead of the id?

`array<ProcessingStep> processingSteps = [];`